providers/kit/metadata.go

```diff
@@ -0,0 +1,90 @@
+package kit
+
+import (
+	"context"
+
+	"github.com/amp-labs/connectors/common"
+	"github.com/amp-labs/connectors/common/jsonquery"
+	"github.com/spyzhov/ajson"
+)
+
+var (
+	perPage          = "per_page" // nolint:gochecknoglobals
+	metadataPageSize = "1"        // nolint:gochecknoglobals
+)
+
+// ListObjectMetadata creates metadata of object via reading objects using Kit API.
+func (c *Connector) ListObjectMetadata(ctx context.Context,
+	objectNames []string,
+) (*common.ListObjectMetadataResult, error) {
+	if len(objectNames) == 0 {
+		return nil, common.ErrMissingObjects
+	}
+
+	metadataResult := common.ListObjectMetadataResult{
+		Result: make(map[string]common.ObjectMetadata),
+		Errors: make(map[string]error),
+	}
+
+	for _, obj := range objectNames {
+		// Constructing the request url.
+		url, err := c.getApiURL(obj)
+		if err != nil {
+			return nil, err
+		}
+
+		url.WithQueryParam(perPage, metadataPageSize)
+
+		resp, err := c.Client.Get(ctx, url.String())
```

Comment:
This approach assumes that every resource returns an array with at least one object.
I don't think that this is a reliable source of Metadata fields.

I tried to search for OpenAPI spec and it is not there, even postman collection linked from their documentation is gone.
The easiest approach would be going over [kit API documentation](https://developers.kit.com/v4#kit-api-accounts) and copying example response to produce the static `schema.json` file manually. There is a possiblity for scrapper if it is easy but I think manual scrapping would be enough.

I think it is important to hear opinion of other reviewers on this matter: @RajatPawar @eberle1080

Reply:
@eberle1080 Do you have any suggestion for this or can I mark it as resolved conversation

Reply:
1. I was working on one connector a while ago, where I noticed an interesting behaviour with some objects. The `test` script started failing for some. And I created this method with a comment: https://github.com/amp-labs/connectors/blob/811c1332919f0782360cdad2dd0414ee3377bb67/test/utils/mockutils/metadataResult.go#L46-L50

**Rule:** `Fields(from read) ⊆ Fields(from metadata)`

Therefore, some fields may not exist at a time of making ListObjectMetadata call. But after API writes as a side effect it will return different, new fields. It seems that OpenAPI (given it is up to date) returns largest possible set of fields.

2. Another point is `dynamicscrm` had an issue when the read response had fields which couldn't be queried and Read would error out.

**Rule:** `Fields(for read query-params) ⊆ Fields(from read resp)`

Therefore, in rare cases not all response fields can be used to construct API call, meaning ListObjectMetadata must return only "safe" fields.